app/models/manageiq/providers/vmware/infra_manager/refresh_parser/filter.rb

```diff
@@ -32,7 +32,7 @@ def filter_vc_data(ems, target)
         vm_data = vm_inv_by_vm(target)
         unless vm_data.nil?
           filtered_data[:vm] = vm_data
-          filtered_data[:storage] = storage_inv_by_vm_inv(vm_data)
+          filtered_data[:storage] = storage_inv_by_host_inv(host_inv_by_host(target.host))
```

Comment:
What if the target doesn't have a host?  What happens then?  Also, this is using "old" data from the database, instead of relying on the "fresh" data from the inventory itself.